subt/artifacts.py

```diff
@@ -436,19 +436,39 @@ class ArtifactReporter(Node):
     def __init__(self, config, bus):
         super().__init__(config, bus)
         bus.register("artf_cmd")
+        self.repeat_report_sec = config.get('repeat_report_sec')
+        self.artf_xyz = []
+        self.artf_xyz_accumulated = []
 
-    def update(self):  # hack, this method should be called run instead!
-        channel = super().update()  # define self.time
-        assert channel == "artf_xyz", channel
-
+    def publish_artf(self, artf_xyz):
         print(self.time, "DETECTED:")
-        for artf_type, ix, iy, iz in self.artf_xyz:
+        for artf_type, ix, iy, iz in artf_xyz:
             print(" ", artf_type, ix/1000.0, iy/1000.0, iz/1000.0)
             s = '%s %.2f %.2f %.2f\n' % (artf_type, ix/1000.0, iy/1000.0, iz/1000.0)
             self.publish('artf_cmd', bytes('artf ' + s, encoding='ascii'))
-
         print('report completed')
 
+    def update(self):  # hack, this method should be called run instead!
+        channel = super().update()  # define self.time
+        assert channel in ["artf_xyz", "sim_time_sec"], channel
+
+        if channel == 'sim_time_sec':
+            if self.repeat_report_sec is None or self.sim_time_sec % self.repeat_report_sec != 0:
+                return channel
+            if len(self.artf_xyz) == 0:
+                return channel
+        elif channel == 'artf_xyz':
+            for item in self.artf_xyz:
+                if item not in self.artf_xyz_accumulated:
+                    self.artf_xyz_accumulated.append(item)
+
+        if self.repeat_report_sec is None:
+            self.publish_artf(self.artf_xyz)
+        else:
+            self.publish_artf(self.artf_xyz_accumulated)
```

Comment:
When I first saw this part, my first question was: why is it here, outside of the ifs? Only when I started writing this comment, I've found the assert on channels. So, overall, it most likely does what was intended, however it is not immediately obvious what that intention was. Maybe adding some comments would help? I am not sure.

Reply:
I actually changed this part later on, that accumulated reports are only on the timer, i.e. I am not sure if to leave it as ti is here and revise it in #636

Reply:
Leave it as is. Just consider adding some comments explaining the intention of the code (not necessarily in this PR).